src/nvim/quickfix.c

```diff
@@ -2723,9 +2721,7 @@ static int qf_jump_edit_buffer(qf_info_T *qi, qfline_T *qf_ptr, int forceit, win
     return NOTDONE;
   }
 
-  if (old_qf_curlist != qi->qf_curlist
```

Comment:
any specific reason these conditions are now dead, like some code that was deleted (which could have changed curlist)? mentioning it in the commit message could be useful for future reference.

Reply:
No idea, I didn't think further than "I see code, I delete code". Let me go through each change and see if I can figure out the reason they became obsolete.